com.unity.multiplayer.mlapi/Runtime/SceneManagement/NetworkSceneManager.cs

```diff
@@ -335,93 +349,49 @@ private static void OnSceneUnloadServer(Guid switchSceneGuid)
-                        ulong ownerClientId = reader.ReadUInt64Packed();
-                        bool hasParent = reader.ReadBool();
-                        ulong? parentNetworkId = null;
-
-                        if (hasParent)
-                        {
-                            parentNetworkId = reader.ReadUInt64Packed();
-                        }
-
-                        uint prefabHash = reader.ReadUInt32Packed();
-
-                        Vector3? position = null;
-                        Quaternion? rotation = null;
-                        if (reader.ReadBool())
-                        {
-                            position = new Vector3(reader.ReadSinglePacked(), reader.ReadSinglePacked(), reader.ReadSinglePacked());
-                            rotation = Quaternion.Euler(reader.ReadSinglePacked(), reader.ReadSinglePacked(), reader.ReadSinglePacked());
-                        }
-
-                        var networkObject = NetworkManager.Singleton.SpawnManager.CreateLocalNetworkObject(false, 0, prefabHash, ownerClientId, parentNetworkId, position, rotation);
-                        NetworkManager.Singleton.SpawnManager.SpawnNetworkObjectLocally(networkObject, networkId, true, isPlayerObject, ownerClientId, objectStream, false, 0, true, false);
-
-                        var bufferQueue = NetworkManager.Singleton.BufferManager.ConsumeBuffersForNetworkId(networkId);
-
-                        // Apply buffered messages
-                        if (bufferQueue != null)
-                        {
-                            while (bufferQueue.Count > 0)
-                            {
-                                BufferManager.BufferedMessage message = bufferQueue.Dequeue();
-                                NetworkManager.Singleton.HandleIncomingData(message.SenderClientId, message.NetworkChannel, new ArraySegment<byte>(message.NetworkBuffer.GetBuffer(), (int)message.NetworkBuffer.Position, (int)message.NetworkBuffer.Length), message.ReceiveTime, false);
-                                BufferManager.RecycleConsumedBufferedMessage(message);
-                            }
-                        }
+                        parentNetworkId = reader.ReadUInt32Packed();
                     }
-                }
-            }
-            else
-            {
-                var networkObjects = UnityEngine.Object.FindObjectsOfType<NetworkObject>();
-                NetworkManager.Singleton.SpawnManager.ClientCollectSoftSyncSceneObjectSweep(networkObjects);
 
-                using (var reader = PooledNetworkReader.Get(objectStream))
-                {
-                    uint newObjectsCount = reader.ReadUInt32Packed();
+                    var prefabHash = reader.ReadUInt32Packed();
+                    Vector3? position = null;
+                    Quaternion? rotation = null;
 
-                    for (int i = 0; i < newObjectsCount; i++)
+                    if (reader.ReadBool())
```

Comment:
comment what this `bool` is? (maybe)

Reply:
Added:
//Check to see if we have position and rotation values that follow
Even though it is relatively obvious that if it is true there is position and rotation information and if false there is not.

Reply:
so that `bool` flag is unnamed at the place where we write on the other hand?

Reply:
Look at lines 293 to 324... that is the server side and it states what is being written.

Reply:
oh, I see.